### PR TITLE
Clean ups for introducing BE DAI widget ops for HDA DAIs

### PR DESCRIPTION
**Pull request by ranj063**
these are the clean up  patches from #3972

### CODE REVIEW

**Review by plbossart — APPROVED** [review]

**Comment by ranj063** [comment]
SOFCI TEST

**Comment by fredoh9** [comment]
CML_HEL_RT5682 has I2C communication timed out. After reboot, same build in same device works fine. Will report this issue.

```
===========================>>
[    4.649612] kernel: i2c_designware i2c_designware.2: controller timed out
[    4.649676] kernel: rt1011 i2c-10EC1011:00: Device with ID register ffff922d is not rt1011
[    5.673698] kernel: i2c_designware i2c_designware.2: controller timed out
[    6.697617] kernel: i2c_designware i2c_designware.2: controller timed out
[    6.697675] kernel: rt1011 i2c-10EC1011:01: Device with ID register ffff922d is not rt1011
[    7.721739] kernel: i2c_designware i2c_designware.2: controller timed out
[    7.721822] kernel: rt5682 i2c-10EC5682:00: Device with ID register ffffffff is not rt5682
[    8.745552] kernel: i2c_designware i2c_designware.2: controller timed out
[    8.745569] kernel: rt1011 i2c-10EC1011:02: Device with ID register ffff922d is not rt1011
[    9.769540] kernel: i2c_designware i2c_designware.2: controller timed out
[    9.769555] kernel: rt1011 i2c-10EC1011:03: Device with ID register ffff922d is not rt1011
<<===========================
```

https://sof-ci.01.org/linuxpr/PR4012/build1916/devicetest/index.html

**Review by RanderWang — APPROVED** [review]
LGTM

File: sound/soc/sof/intel/hda-dai.c [hunk 1]
```diff
@@ -242,7 +214,28 @@ static int hda_link_dma_hw_params(struct snd_pcm_substream *substream,
...
+	dev_dbg(bus->dev, "format_val=%d, rate=%d, ch=%d, format=%d\n",
+		format_val, p_params.s_freq, p_params.ch, p_params.format);
+
+	snd_hdac_ext_stream_setup(hext_stream, format_val);
+
+	if (hext_stream->hstream.direction == SNDRV_PCM_STREAM_PLAYBACK) {
+		list_for_each_entry(hlink, &bus->hlink_list, list) {
+			if (hlink->index == p_params.link_index)
+				snd_hdac_ext_bus_link_set_stream_id(hlink,
+								    stream_tag);
```
Comment (ujfalusi) [inline sound/soc/sof/intel/hda-dai.c]: nitpick: it can fit to one line (88 char long line)

File: sound/soc/sof/intel/hda-dai.c [hunk 2]
```diff
@@ -225,6 +195,8 @@ static int hda_link_dma_hw_params(struct snd_pcm_substream *substream,
 		snd_soc_dai_set_dma_data(cpu_dai, substream, (void *)hext_stream);
 	}
 
+	hstream = &hext_stream->hstream;
+	stream_tag = hstream->stream_tag;
```
Comment (ujfalusi) [inline sound/soc/sof/intel/hda-dai.c:186]: You don't really need the `hstream` here:
```c
stream_tag = hext_stream->hstream.stream_tag;
```
or use it at line 228:
```c
if (hstream->direction == SNDRV_PCM_STREAM_PLAYBACK) {
```

File: sound/soc/sof/intel/hda-dai.c [hunk 3]
```diff
@@ -201,38 +193,27 @@ static int hda_link_dma_hw_params(struct snd_pcm_substream *substream,
 	if (!hlink)
 		return -EINVAL;
 
+	if (hext_stream->hstream.direction == SNDRV_PCM_STREAM_PLAYBACK)
+		snd_hdac_ext_bus_link_set_stream_id(hlink, stream_tag);
+
```
Comment (ujfalusi) [inline sound/soc/sof/intel/hda-dai.c:193]: you could mention that the order of configuration is changing:
now you configure the stream_id before stream_reset.

Is there a reason for the sequence change?

**Review by bardliao — APPROVED** [review]

File: sound/soc/sof/intel/hda-dai.c (hunk 2 above)
Comment (ranj063) [reply sound/soc/sof/intel/hda-dai.c:186]: can we let this one go? all this code will eventually go away with the ops anyway

File: sound/soc/sof/intel/hda-dai.c (hunk 1 above)
Comment (ranj063) [reply sound/soc/sof/intel/hda-dai.c]: same here, this goes away in the very next patch

File: sound/soc/sof/intel/hda-dai.c (hunk 3 above)
Comment (ranj063) [reply sound/soc/sof/intel/hda-dai.c:193]: the main reason is to club the use of hlink together so that this can be moved to the op later on.

Comment (plbossart) [reply sound/soc/sof/intel/hda-dai.c:193]: @ujfalusi @bardliao there are two independent parts here:

snd_hdac_ext_bus_link_set_stream_id(hlink, stream_tag); configure the multi-link mapping of streams to DMAs.

hdax_ext_stream_reset/setup program the DMA proper.

It's two different levels that can be done in any order. The only required order is reset before setup.

**Comment by plbossart** [comment]
@ujfalusi any objections or can we merge this before the PR #3972?

**Review by ujfalusi — APPROVED** [review]

**Pull request merged**